authlib/integrations/httpx_client/oauth2_client.py

```diff
@@ -150,3 +151,101 @@ def _http_post(self, url, body=None, auth=None, headers=None, **kwargs):
         return self.post(
             url, data=dict(url_decode(body)),
             headers=headers, auth=auth, **kwargs)
+
+class OAuth2Client(_OAuth2Client, Client):
+    SESSION_REQUEST_PARAMS = HTTPX_CLIENT_KWARGS
+
+    client_auth_class = OAuth2ClientAuth
+    token_auth_class = OAuth2Auth
+
+    def __init__(self, client_id=None, client_secret=None,
+                 token_endpoint_auth_method=None,
+                 revocation_endpoint_auth_method=None,
+                 scope=None, redirect_uri=None,
+                 token=None, token_placement='header',
+                 update_token=None, **kwargs):
+
+        # extract httpx.Client kwargs
+        client_kwargs = self._extract_session_request_params(kwargs)
+        Client.__init__(self, **client_kwargs)
+
+        _OAuth2Client.__init__(
+            self, session=None,
+            client_id=client_id, client_secret=client_secret,
+            token_endpoint_auth_method=token_endpoint_auth_method,
+            revocation_endpoint_auth_method=revocation_endpoint_auth_method,
+            scope=scope, redirect_uri=redirect_uri,
+            token=token, token_placement=token_placement,
+            update_token=update_token, **kwargs
+        )
+
+    @staticmethod
+    def handle_error(error_type, error_description):
+        raise OAuthError(error_type, error_description)
+
+    def request(self, method, url, withhold_token=False, auth=None, **kwargs):
+        if not withhold_token and isinstance(auth, UnsetType):
+            if not self.token:
+                raise MissingTokenError()
+
+            if self.token.is_expired():
+                self.ensure_active_token()
+
+            auth = self.token_auth
+
+        return super(OAuth2Client, self).request(
+            method, url, auth=auth, **kwargs)
+
+    def ensure_active_token(self):
+        refresh_token = self.token.get('refresh_token')
+        url = self.metadata.get('token_endpoint')
+        if refresh_token and url:
+            self.refresh_token(url, refresh_token=refresh_token)
+        elif self.metadata.get('grant_type') == 'client_credentials':
+            access_token = self.token['access_token']
+            token = self.fetch_token(url, grant_type='client_credentials')
+            if self.update_token:
+                self.update_token(token, access_token=access_token)
+        else:
+            raise InvalidTokenError()
+
+    def _fetch_token(self, url, body='', headers=None, auth=None,
```

Comment:
You can check the implementation of `requests`, there is no need to implemented it if you have set `session=self`.